libraries/AP_ADSB/AP_ADSB.cpp

```diff
@@ -259,30 +272,45 @@ void AP_ADSB::detect_instance(uint8_t instance)
     case Type::uAvionix_MAVLink:
 #if HAL_ADSB_UAVIONIX_MAVLINK_ENABLED
         if (AP_ADSB_uAvionix_MAVLink::detect()) {
-            _backend[instance] = new AP_ADSB_uAvionix_MAVLink(*this, instance);
-            return;
+            _backend[instance] = new AP_ADSB_uAvionix_MAVLink(*this, state[instance], instance);
+            // return;
```

Comment:
We try to not carry dead code, so we will need those removed if we don't use them anymore